src/components/dao/dao.js

```diff
@@ -325,44 +325,52 @@ export default class DaoComponent extends Component {
         let newState = {};
 
         // initialize DAO object
-        SputnikDAO.init(dao.address).catch(e => {})
+        SputnikDAO.init(daoAddress).catch(e => {})
         .then((newDAO) => {
+            this.setState({ dao: newDAO });
             // DAO not ready => either no SputnikDAO contract on the chosen address
             // or some error happened during DAO object init.
-            this.setState({ dao: newDAO });
-            if (!newDAO.ready) { return; }
-        }).then(() => {
-            return Promise.all([
-                view(multicall, "get_admins", {}).catch(e => {
-                        if (e.type === "AccountDoesNotExist" && e.toString().includes(` ${multicall} `)) {
-                            noContract.isBad = true;
+            if (!newDAO.ready) {
+                noContract.isBad = true;
+                noDao.isBad = true;
+                this.setState({ loading: false });
+                return;
+            }
+            // DAO correctly initialized, try to fetch multicall info
+            else {
```

Comment:
No need for `else` if the "then" part already `return`s

Reply:
Just in case the `return;` is removed in the future